### PR TITLE
Send epoch timestamp with the frame

### LINKED ISSUE #36: Compute timestamps in wait_for_animation_frame
Every frame we just increment the timestamp by 1.0

https://github.com/servo/webxr/blob/fcbdddb66e2d307d86ff109d79001c7a45de79db/webxr-api/session.rs#L200

We should instead make it part of `wait_for_animation_frame`'s return value. Some devices (e.g. OpenXR) provide the current frame timestamp directly; for others we can query system time APIs.

cc @jdm

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes #36

r? @asajeffrey

### CODE REVIEW

File: webxr/openxr/mod.rs [hunk 1]
```diff
@@ -447,6 +448,7 @@ impl DeviceAPI<Surface> for OpenXrDevice {
             transform,
             inputs: vec![right_input_frame, left_input_frame],
             events,
+            time_ns
```
Comment (jdm) [inline webxr/openxr/mod.rs]: Shouldn't this come from https://docs.rs/openxr/0.10.0/openxr/struct.FrameState.html#structfield.predicted_display_time instead?

Comment (Manishearth) [reply webxr/openxr/mod.rs]: That's the predicted display time, not the time for the actual frame data.

That said the frame data may itself be predicted. I'm not sure. There are WPT tests ensuring that Performance.now() is greater than the frame timestamp.

Comment (Manishearth) [reply webxr/openxr/mod.rs]: Basically, I'm not sure what to do here and need spec clarification

File: webxr-api/frame.rs [hunk 2]
```diff
@@ -26,6 +26,9 @@ pub struct Frame {
 
     /// Events that occur with the frame.
     pub events: Vec<FrameUpdateEvent>,
+
+    /// Value of time::precise_time_ns() when frame was obtained
+    pub time_ns: u64,
```
Comment (asajeffrey) [inline webxr-api/frame.rs]: Use an `Instant` here?

Comment (Manishearth) [reply webxr-api/frame.rs]: Instant can only be compared with other Instants, and we don't use that in servo for `Performance.now()`

**Pull request merged**